lib/collection/src/shards/local_shard.rs

```diff
@@ -819,15 +805,17 @@ impl LocalShard {
     }
 }
 
-pub async fn drop_and_delete_from_disk(shard: LocalShard) -> CollectionResult<()> {
-    let path = shard.shard_path();
-    drop(shard);
-    remove_dir_all(path).await?;
-    Ok(())
-}
-
 impl Drop for LocalShard {
     fn drop(&mut self) {
-        self.assert_before_drop_called()
+        thread::scope(|s| {
+            let handle = thread::Builder::new()
+                .name("drop-shard".to_string())
+                .spawn_scoped(s, || {
+                    // Needs dedicated thread to avoid `Cannot start a runtime from within a runtime` error.
+                    self.update_runtime
+                        .block_on(async { self.stop_gracefully().await })
+                });
+            handle.expect("Failed to create thread for shard drop");
+        })
```

Comment:
Can't we do this without the scoped thread? I believe the end result is the same, and would prevent spawning an extra thread.

Reply:
Your comment explains: _Needs dedicated thread to avoid `Cannot start a runtime from within a runtime` error._

I'll just give simplifying this a bit a shot.

Reply:
The part behind it explains a little better why we _need_ this thread:

> This happens because a function (like `block_on`) attempted to block the current thread while the thread is being used to drive asynchronous tasks.